src/main/java/net/sf/jabref/logic/groups/GroupTreeNode.java

```diff
@@ -213,4 +213,20 @@ public String toString() {
     public GroupTreeNode copyNode() {
         return new GroupTreeNode(group);
     }
+
+    /**
+     * Determines the number of entries in the specified list which are matched by this group.
+     * @param entries list of entries to be searched
+     * @return number of hits
+     */
+    public int numberOfHits(List<BibEntry> entries) {
+        int hits = 0;
+        SearchMatcher matcher = getSearchRule();
+        for (BibEntry entry : entries) {
+            if (matcher.isMatch(entry)) {
+                hits++;
+            }
+        }
+        return hits;
+    }
```

Comment:
The (Abstract)Group does not know where it is sitting in the group tree but the GroupTreeNode does. But the position in the tree might influence the search result (since we support refining / independent / ... search modes)